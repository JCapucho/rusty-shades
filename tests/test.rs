use rusty_shades::build_naga_ir;
use std::fs::read_to_string;

#[test]
fn insta_test() {
    insta::glob!("*.rsh", |file| {
        let code = read_to_string(file).unwrap();
        let res = build_naga_ir(&code);
        insta::assert_ron_snapshot!(res);
    });
}
