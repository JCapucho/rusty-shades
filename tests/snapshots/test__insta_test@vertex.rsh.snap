---
source: tests/test.rs
expression: res
input_file: tests/vertex.rsh
---
Ok(Module(
  header: Header(
    version: (1, 0, 0),
    generator: 1920299892,
  ),
  types: [
    Type(
      name: None,
      inner: Matrix(
        columns: Quad,
        rows: Quad,
        width: 4,
      ),
    ),
    Type(
      name: None,
      inner: Vector(
        size: Tri,
        kind: Float,
        width: 4,
      ),
    ),
    Type(
      name: Some("Globals"),
      inner: Struct(
        members: [
          StructMember(
            name: Some("view_proj"),
            origin: Offset(0),
            ty: 1,
          ),
          StructMember(
            name: Some("view_position"),
            origin: Offset(4),
            ty: 2,
          ),
        ],
      ),
    ),
    Type(
      name: None,
      inner: Vector(
        size: Bi,
        kind: Float,
        width: 4,
      ),
    ),
    Type(
      name: Some("Locals"),
      inner: Struct(
        members: [
          StructMember(
            name: Some("transform"),
            origin: Offset(0),
            ty: 1,
          ),
          StructMember(
            name: Some("min_max"),
            origin: Offset(4),
            ty: 4,
          ),
        ],
      ),
    ),
    Type(
      name: Some("Globals"),
      inner: Struct(
        members: [
          StructMember(
            name: Some("view_proj"),
            origin: Offset(0),
            ty: 1,
          ),
          StructMember(
            name: Some("view_position"),
            origin: Offset(4),
            ty: 2,
          ),
        ],
      ),
    ),
    Type(
      name: Some("Locals"),
      inner: Struct(
        members: [
          StructMember(
            name: Some("transform"),
            origin: Offset(0),
            ty: 1,
          ),
          StructMember(
            name: Some("min_max"),
            origin: Offset(4),
            ty: 4,
          ),
        ],
      ),
    ),
    Type(
      name: None,
      inner: Vector(
        size: Quad,
        kind: Float,
        width: 4,
      ),
    ),
    Type(
      name: None,
      inner: Scalar(
        kind: Float,
        width: 4,
      ),
    ),
  ],
  constants: [
    Constant(
      name: None,
      specialization: None,
      inner: Float(1),
      ty: 9,
    ),
  ],
  global_variables: [
    GlobalVariable(
      name: Some("a_position"),
      class: Input,
      binding: Some(Location(0)),
      ty: 2,
      interpolation: None,
      storage_access: StorageAccess(
        bits: 0,
      ),
    ),
    GlobalVariable(
      name: Some("a_color"),
      class: Input,
      binding: Some(Location(1)),
      ty: 2,
      interpolation: None,
      storage_access: StorageAccess(
        bits: 0,
      ),
    ),
    GlobalVariable(
      name: Some("a_normal"),
      class: Input,
      binding: Some(Location(2)),
      ty: 2,
      interpolation: None,
      storage_access: StorageAccess(
        bits: 0,
      ),
    ),
    GlobalVariable(
      name: Some("v_position"),
      class: Output,
      binding: Some(Location(0)),
      ty: 2,
      interpolation: None,
      storage_access: StorageAccess(
        bits: 0,
      ),
    ),
    GlobalVariable(
      name: Some("v_color"),
      class: Output,
      binding: Some(Location(1)),
      ty: 2,
      interpolation: None,
      storage_access: StorageAccess(
        bits: 0,
      ),
    ),
    GlobalVariable(
      name: Some("v_normal"),
      class: Output,
      binding: Some(Location(2)),
      ty: 2,
      interpolation: None,
      storage_access: StorageAccess(
        bits: 0,
      ),
    ),
    GlobalVariable(
      name: Some("gl_position"),
      class: Output,
      binding: Some(BuiltIn(Position)),
      ty: 8,
      interpolation: None,
      storage_access: StorageAccess(
        bits: 0,
      ),
    ),
    GlobalVariable(
      name: Some("gl_position"),
      class: Input,
      binding: Some(BuiltIn(Position)),
      ty: 8,
      interpolation: None,
      storage_access: StorageAccess(
        bits: 0,
      ),
    ),
    GlobalVariable(
      name: Some("globals"),
      class: Uniform,
      binding: Some(Resource(
        group: 0,
        binding: 0,
      )),
      ty: 3,
      interpolation: None,
      storage_access: StorageAccess(
        bits: 0,
      ),
    ),
    GlobalVariable(
      name: Some("locals"),
      class: Uniform,
      binding: Some(Resource(
        group: 2,
        binding: 0,
      )),
      ty: 5,
      interpolation: None,
      storage_access: StorageAccess(
        bits: 0,
      ),
    ),
  ],
  functions: [],
  entry_points: {
    (Vertex, "vertex_main"): EntryPoint(
      early_depth_test: None,
      workgroup_size: (0, 0, 0),
      function: Function(
        name: None,
        parameter_types: [],
        return_type: None,
        global_usage: [
          GlobalUse(
            bits: 1,
          ),
          GlobalUse(
            bits: 1,
          ),
          GlobalUse(
            bits: 1,
          ),
          GlobalUse(
            bits: 2,
          ),
          GlobalUse(
            bits: 2,
          ),
          GlobalUse(
            bits: 2,
          ),
          GlobalUse(
            bits: 2,
          ),
          GlobalUse(
            bits: 0,
          ),
          GlobalUse(
            bits: 1,
          ),
          GlobalUse(
            bits: 1,
          ),
        ],
        local_variables: [
          LocalVariable(
            name: None,
            ty: 2,
            init: None,
          ),
          LocalVariable(
            name: None,
            ty: 2,
            init: None,
          ),
        ],
        expressions: [
          GlobalVariable(5),
          GlobalVariable(2),
          GlobalVariable(6),
          GlobalVariable(3),
          LocalVariable(1),
          GlobalVariable(1),
          GlobalVariable(4),
          GlobalVariable(10),
          LocalVariable(1),
          AccessIndex(
            base: 9,
            index: 0,
          ),
          LocalVariable(1),
          AccessIndex(
            base: 11,
            index: 1,
          ),
          LocalVariable(1),
          AccessIndex(
            base: 13,
            index: 2,
          ),
          Constant(1),
          AccessIndex(
            base: 8,
            index: 0,
          ),
          Compose(
            ty: 8,
            components: [
              10,
              12,
              14,
              15,
            ],
          ),
          Binary(
            op: Multiply,
            left: 16,
            right: 17,
          ),
          AccessIndex(
            base: 18,
            index: 0,
          ),
          AccessIndex(
            base: 18,
            index: 1,
          ),
          AccessIndex(
            base: 18,
            index: 2,
          ),
          Compose(
            ty: 2,
            components: [
              19,
              20,
              21,
            ],
          ),
          LocalVariable(2),
          GlobalVariable(1),
          GlobalVariable(7),
          GlobalVariable(9),
          GlobalVariable(10),
          AccessIndex(
            base: 26,
            index: 0,
          ),
          AccessIndex(
            base: 27,
            index: 0,
          ),
          LocalVariable(2),
          AccessIndex(
            base: 30,
            index: 0,
          ),
          LocalVariable(2),
          AccessIndex(
            base: 32,
            index: 1,
          ),
          LocalVariable(2),
          AccessIndex(
            base: 34,
            index: 2,
          ),
          Constant(1),
          Binary(
            op: Multiply,
            left: 28,
            right: 29,
          ),
          Compose(
            ty: 8,
            components: [
              31,
              33,
              35,
              36,
            ],
          ),
          Binary(
            op: Multiply,
            left: 37,
            right: 38,
          ),
        ],
        body: [
          Store(
            pointer: 1,
            value: 2,
          ),
          Store(
            pointer: 3,
            value: 4,
          ),
          Store(
            pointer: 5,
            value: 6,
          ),
          Store(
            pointer: 7,
            value: 22,
          ),
          Store(
            pointer: 23,
            value: 24,
          ),
          Store(
            pointer: 25,
            value: 39,
          ),
        ],
      ),
    ),
  },
))
